src/tooltip-manager.ts

```diff
@@ -0,0 +1,137 @@
+'use babel';
+
+/// <reference path="marked/index.d.ts" />
+
+import {CompositeDisposable} from 'atom';
+import {RemoteEditorService} from 'polymer-analyzer/lib/editor-service/remote-editor-service';
+import * as marked from 'marked';
+
+class TooltipManager {
+  textEditor: AtomCore.IEditor;
+  tooltipMarker: AtomCore.IDisplayBufferMarker;
+  tooltipElement: HTMLElement = null;
+  documentationElement: HTMLElement = null;
+  editorService: RemoteEditorService;
+  editorElements: Array<HTMLElement> = [];
+  cursorOutsideTooltip: boolean = true;
+
+  constructor(editorService: RemoteEditorService) {
+    this.editorService = editorService;
+  };
+
+  subscribe(subscriptions: CompositeDisposable) {
+    subscriptions.add(atom.workspace.observePanes((pane: AtomCore.IPane) => {
+      this.update();
+      subscriptions.add(pane.onDidChangeActiveItem(() => {
+        this.update();
+      }));
+    }));
+  };
+
+  update() {
+    this.removeTooltip();
+    this.textEditor = atom.workspace.getActiveTextEditor();
+    this.addMouseEventListener();
+  }
+
+  addMouseEventListener() {
+    const textEditorElement = atom.views.getView(this.textEditor);
+    if (!textEditorElement) {
+      return;
+    }
+    if (this.editorElements.indexOf(textEditorElement) > -1) {
+      return;
+    }
+    this.editorElements.push(textEditorElement);
+    let timer: number;
+    textEditorElement.addEventListener('mousemove', (event: MouseEvent) => {
```

Comment:
We should probably remove this event listener when the active editor changes. Maybe something like:

```typescript
if (this.removeMouseMoveListener) {
  this.removeMouseMoveListener();
}
const mouseMoveListener = (event: MouseEvent) => { ... };
textEditorEelement.addEventListener('mousemove', mouseMoveListener);
this.removeMouseMoveListener = () => textEditorElement.removeEventListener('mousemove', mouseMoveListener);
```